pkg/cloudprovider/provider/openstack/helper.go

```diff
@@ -361,6 +365,15 @@ func assignFloatingIPToInstance(client *gophercloud.ProviderClient, instanceID,
 		if ip, err = createFloatingIP(client, region, port.ID, floatingIPPool); err != nil {
 			return osErrorToTerminalError(err, "failed to allocate a floating ip")
 		}
+		if _, err = machineUpdater(machine, func(m *v1alpha1.Machine) {
```

Comment:
Move the update of the machine to the `provider.go`.
`helper.go` so far only offered helper functions around the Gophercloud SDK - and that's something that should remain - mainly for `separation of concern`.

Reply:
Done